src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,178 @@
+open Core
+open Integration_test_lib
+open Signature_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let block_producer_balance = "1000" (* 1_000_000_000_000 *)
+
+  let config =
+    let n = 3 in
```

Comment:
@imeckler lmk if it's better to raise the number of block producers, i've reduced it just to iterate faster but can raise it again